src/pf-tooltip/pf-tooltip.component.js

```diff
@@ -0,0 +1,304 @@
+import {default as tmpl} from 'pf-tooltip.template';
+import {pfUtil} from 'pf-utils';
+
+/**
+ * <b>&lt;pf-tooltip&gt;</b> element for Patternfly Web Components
+ *
+ * @example {@lang xml}
+ * <pf-tooltip animation="fade" target="#btn-left" placement="left" delay="100" duration="150" container="#container"></pf-alert>
+ *
+ * @prop {string} animation the animation class
+ * @prop {string} target the target element selector
+ * @prop {string} placement left, right, top, bottom
+ * @prop {number} delay animation delay (ms)
+ * @prop {number} duration animation duration (ms)
+ * @prop {string} container the container element selector
+ */
+
+export class PfTooltip extends HTMLElement {
+
+  /**
+   * Reinitializes tooltip component with attribute values and resets content
+   */
+  init () {
+    this.element = this;
+    this.content = this.element.innerHTML;
+    this.tooltip = null;
+    this._target = this.getAttribute('target') ? document.querySelector(this.getAttribute('target')) : this;
```

Comment:
I'm thinking we should use the attribute name targetSelector? Looking at the attribute only, it's not clear that the target accepts an ID, class, etc.

I see "selector" mentioned in the JsDoc, but making the attribute names clear would be helpful. The datatables.net attributes append "Selector" to the attributes, for example. 

It would help folks understand that they need to use "#btn-left" instead of "btn-left", for example.